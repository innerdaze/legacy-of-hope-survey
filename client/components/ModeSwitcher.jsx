import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Menu from 'grommet/components/Menu'
import Anchor from 'grommet/components/Anchor'
import Header from 'grommet/components/Header'
import Modes from '../constants/OperationModes'

class ModeSwitcher extends Component {
  constructor(props) {
    super(props)

    this.state = {
      mode: Modes.STOCKTAKE
    }
  }

  onSwitch(mode) {
    this.setState({
      mode: mode
    })

    this.props.onSwitch(mode)
  }

  render() {
    return (
      <Menu primary={true}>
        <Anchor
          label='Stocktake'
          onClick={this.onSwitch.bind(this, Modes.STOCKTAKE)}
          className={this.state.mode === Modes.STOCKTAKE && 'active'}
          />
        <Anchor
          label='Order'
          onClick={this.onSwitch.bind(this, Modes.ORDERING)}
          className={this.state.mode === Modes.ORDERING && 'active'}
          />
        <Anchor
          label='Delivery'
          onClick={this.onSwitch.bind(this, Modes.DELIVERY)}
          className={this.state.mode === Modes.DELIVERY && 'active'}
          />
      </Menu>
    )
  }
}

ModeSwitcher.propTypes = {
  onSwitch: PropTypes.func.isRequired
}

ModeSwitcher.defaultTypes = {
  onSwitch: Function.prototype
}

export default ModeSwitcher
