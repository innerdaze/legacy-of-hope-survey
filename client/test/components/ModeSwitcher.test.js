import React from 'react'
import ModeSwitcher from '../../components/ModeSwitcher'

let wrapper, onSwitch, activeMode

describe('Testing on ModeSwitcher...', () => {
  beforeEach(() => {
    onSwitch = jest.fn()
    activeMode = 0
    wrapper = shallow(
      <ModeSwitcher onSwitch={onSwitch} activeMode={activeMode} />
    )
  })

  describe('ModeSwitcher layout', () => {
    test('Expect ModeSwitcher to have parent Menu comp and 4 Anchor child comp', () => {
      expect(wrapper.find('Menu')).toHaveLength(1)
      expect(wrapper.find('Menu').find('Anchor')).toHaveLength(4)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Test on prop', () => {
    test('Expect onSwitch prop function to called if one of the Anchor is clicked', () => {
      wrapper.find('Anchor[label="Stocktake"]').simulate('click')
      expect(onSwitch).toBeCalled()
      expect(wrapper).toMatchSnapshot()

      wrapper.find('Anchor[label="Order"]').simulate('click')
      expect(onSwitch).toBeCalled()
      expect(wrapper).toMatchSnapshot()

      wrapper.find('Anchor[label="Delivery"]').simulate('click')
      expect(onSwitch).toBeCalled()
      expect(wrapper).toMatchSnapshot()

      wrapper.find('Anchor[label="Shelf Labels"]').simulate('click')
      expect(onSwitch).toBeCalled()
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Test on prop', () => {
    describe('Expect mode state to change and activeMode prop change', () => {
      wrapper = mount(
        <ModeSwitcher onSwitch={onSwitch} activeMode={activeMode} />
      )
      wrapper.setState({
        activeMode: 1
      })
      wrapper.setProps({
        activeMode: 2
      })

      expect(wrapper.state('mode')).toEqual(2)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Test on state', () => {
    describe('Expect mode state to equal actionMode props when first initialized', () => {
      activeMode = 1
      wrapper = mount(
        <ModeSwitcher onSwitch={onSwitch} activeMode={activeMode} />
      )

      expect(wrapper.state('mode')).toEqual(1)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
