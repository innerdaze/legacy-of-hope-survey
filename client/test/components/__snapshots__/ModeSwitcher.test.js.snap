// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<ModeSwitcher
  activeMode={2}
  onSwitch={[Function]}
>
  <Menu
    closeOnClick={true}
    direction="column"
    dropAlign={
      Object {
        "left": "left",
        "top": "top",
      }
    }
    pad="none"
    primary={true}
  >
    <Box
      announce={false}
      className="grommetux-menu grommetux-menu--column grommetux-menu--primary grommetux-menu--inline"
      direction="column"
      focusable={true}
      pad="none"
      primary={false}
      responsive={true}
      tag="nav"
    >
      <nav
        className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-none grommetux-menu grommetux-menu--column grommetux-menu--primary grommetux-menu--inline"
        style={Object {}}
      >
        <Anchor
          animateIcon={true}
          className={false}
          label="Stocktake"
          method="push"
          onClick={[Function]}
          tag="a"
        >
          <a
            className="grommetux-anchor"
            onClick={[Function]}
          >
            Stocktake
          </a>
        </Anchor>
        <Anchor
          animateIcon={true}
          className={false}
          label="Order"
          method="push"
          onClick={[Function]}
          tag="a"
        >
          <a
            className="grommetux-anchor"
            onClick={[Function]}
          >
            Order
          </a>
        </Anchor>
        <Anchor
          animateIcon={true}
          className={false}
          label="Delivery"
          method="push"
          onClick={[Function]}
          tag="a"
        >
          <a
            className="grommetux-anchor"
            onClick={[Function]}
          >
            Delivery
          </a>
        </Anchor>
        <Anchor
          animateIcon={true}
          className={false}
          label="Shelf Labels"
          method="push"
          onClick={[Function]}
          tag="a"
        >
          <a
            className="grommetux-anchor"
            onClick={[Function]}
          >
            Shelf Labels
          </a>
        </Anchor>
      </nav>
    </Box>
  </Menu>
</ModeSwitcher>
`;

exports[` 2`] = `
<ModeSwitcher
  activeMode={1}
  onSwitch={[Function]}
>
  <Menu
    closeOnClick={true}
    direction="column"
    dropAlign={
      Object {
        "left": "left",
        "top": "top",
      }
    }
    pad="none"
    primary={true}
  >
    <Box
      announce={false}
      className="grommetux-menu grommetux-menu--column grommetux-menu--primary grommetux-menu--inline"
      direction="column"
      focusable={true}
      pad="none"
      primary={false}
      responsive={true}
      tag="nav"
    >
      <nav
        className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-none grommetux-menu grommetux-menu--column grommetux-menu--primary grommetux-menu--inline"
        style={Object {}}
      >
        <Anchor
          animateIcon={true}
          className={false}
          label="Stocktake"
          method="push"
          onClick={[Function]}
          tag="a"
        >
          <a
            className="grommetux-anchor"
            onClick={[Function]}
          >
            Stocktake
          </a>
        </Anchor>
        <Anchor
          animateIcon={true}
          className={false}
          label="Order"
          method="push"
          onClick={[Function]}
          tag="a"
        >
          <a
            className="grommetux-anchor"
            onClick={[Function]}
          >
            Order
          </a>
        </Anchor>
        <Anchor
          animateIcon={true}
          className={false}
          label="Delivery"
          method="push"
          onClick={[Function]}
          tag="a"
        >
          <a
            className="grommetux-anchor"
            onClick={[Function]}
          >
            Delivery
          </a>
        </Anchor>
        <Anchor
          animateIcon={true}
          className={false}
          label="Shelf Labels"
          method="push"
          onClick={[Function]}
          tag="a"
        >
          <a
            className="grommetux-anchor"
            onClick={[Function]}
          >
            Shelf Labels
          </a>
        </Anchor>
      </nav>
    </Box>
  </Menu>
</ModeSwitcher>
`;

exports[`Testing on ModeSwitcher... ModeSwitcher layout Expect ModeSwitcher to have parent Menu comp and 4 Anchor child comp 1`] = `
<Menu
  closeOnClick={true}
  direction="column"
  dropAlign={
    Object {
      "left": "left",
      "top": "top",
    }
  }
  pad="none"
  primary={true}
>
  <Anchor
    animateIcon={true}
    className={false}
    label="Stocktake"
    method="push"
    onClick={[Function]}
    tag="a"
  />
  <Anchor
    animateIcon={true}
    className={false}
    label="Order"
    method="push"
    onClick={[Function]}
    tag="a"
  />
  <Anchor
    animateIcon={true}
    className={false}
    label="Delivery"
    method="push"
    onClick={[Function]}
    tag="a"
  />
  <Anchor
    animateIcon={true}
    className={false}
    label="Shelf Labels"
    method="push"
    onClick={[Function]}
    tag="a"
  />
</Menu>
`;

exports[`Testing on ModeSwitcher... Test on prop Expect onSwitch prop function to called if one of the Anchor is clicked 1`] = `
<Menu
  closeOnClick={true}
  direction="column"
  dropAlign={
    Object {
      "left": "left",
      "top": "top",
    }
  }
  pad="none"
  primary={true}
>
  <Anchor
    animateIcon={true}
    className="active"
    label="Stocktake"
    method="push"
    onClick={[Function]}
    tag="a"
  />
  <Anchor
    animateIcon={true}
    className={false}
    label="Order"
    method="push"
    onClick={[Function]}
    tag="a"
  />
  <Anchor
    animateIcon={true}
    className={false}
    label="Delivery"
    method="push"
    onClick={[Function]}
    tag="a"
  />
  <Anchor
    animateIcon={true}
    className={false}
    label="Shelf Labels"
    method="push"
    onClick={[Function]}
    tag="a"
  />
</Menu>
`;

exports[`Testing on ModeSwitcher... Test on prop Expect onSwitch prop function to called if one of the Anchor is clicked 2`] = `
<Menu
  closeOnClick={true}
  direction="column"
  dropAlign={
    Object {
      "left": "left",
      "top": "top",
    }
  }
  pad="none"
  primary={true}
>
  <Anchor
    animateIcon={true}
    className={false}
    label="Stocktake"
    method="push"
    onClick={[Function]}
    tag="a"
  />
  <Anchor
    animateIcon={true}
    className="active"
    label="Order"
    method="push"
    onClick={[Function]}
    tag="a"
  />
  <Anchor
    animateIcon={true}
    className={false}
    label="Delivery"
    method="push"
    onClick={[Function]}
    tag="a"
  />
  <Anchor
    animateIcon={true}
    className={false}
    label="Shelf Labels"
    method="push"
    onClick={[Function]}
    tag="a"
  />
</Menu>
`;

exports[`Testing on ModeSwitcher... Test on prop Expect onSwitch prop function to called if one of the Anchor is clicked 3`] = `
<Menu
  closeOnClick={true}
  direction="column"
  dropAlign={
    Object {
      "left": "left",
      "top": "top",
    }
  }
  pad="none"
  primary={true}
>
  <Anchor
    animateIcon={true}
    className={false}
    label="Stocktake"
    method="push"
    onClick={[Function]}
    tag="a"
  />
  <Anchor
    animateIcon={true}
    className={false}
    label="Order"
    method="push"
    onClick={[Function]}
    tag="a"
  />
  <Anchor
    animateIcon={true}
    className="active"
    label="Delivery"
    method="push"
    onClick={[Function]}
    tag="a"
  />
  <Anchor
    animateIcon={true}
    className={false}
    label="Shelf Labels"
    method="push"
    onClick={[Function]}
    tag="a"
  />
</Menu>
`;

exports[`Testing on ModeSwitcher... Test on prop Expect onSwitch prop function to called if one of the Anchor is clicked 4`] = `
<Menu
  closeOnClick={true}
  direction="column"
  dropAlign={
    Object {
      "left": "left",
      "top": "top",
    }
  }
  pad="none"
  primary={true}
>
  <Anchor
    animateIcon={true}
    className={false}
    label="Stocktake"
    method="push"
    onClick={[Function]}
    tag="a"
  />
  <Anchor
    animateIcon={true}
    className={false}
    label="Order"
    method="push"
    onClick={[Function]}
    tag="a"
  />
  <Anchor
    animateIcon={true}
    className={false}
    label="Delivery"
    method="push"
    onClick={[Function]}
    tag="a"
  />
  <Anchor
    animateIcon={true}
    className="active"
    label="Shelf Labels"
    method="push"
    onClick={[Function]}
    tag="a"
  />
</Menu>
`;
